src/renderer/wg_engine/tvgWgRenderer.cpp

```diff
@@ -145,31 +132,26 @@ RenderData WgRenderer::prepare(const RenderShape& rshape, RenderData data, const
     if (!renderDataShape) {
         renderDataShape = new WgRenderDataShape();
         renderDataShape->initialize(mDevice);
-        renderDataShape->mRenderSettingsShape.mPipelineBindGroupSolid.initialize(mDevice, mPipelineSolid);
-        renderDataShape->mRenderSettingsShape.mPipelineBindGroupLinear.initialize(mDevice, mPipelineLinear);
-        renderDataShape->mRenderSettingsShape.mPipelineBindGroupRadial.initialize(mDevice, mPipelineRadial);
-        renderDataShape->mRenderSettingsStroke.mPipelineBindGroupSolid.initialize(mDevice, mPipelineSolid);
-        renderDataShape->mRenderSettingsStroke.mPipelineBindGroupLinear.initialize(mDevice, mPipelineLinear);
-        renderDataShape->mRenderSettingsStroke.mPipelineBindGroupRadial.initialize(mDevice, mPipelineRadial);
     }
     
-    if (flags & (RenderUpdateFlag::Path | RenderUpdateFlag::Stroke))
+    // update geometry
+    if (flags & (RenderUpdateFlag::Path | RenderUpdateFlag::Stroke)) {
         renderDataShape->releaseRenderData();
-
-    if (flags & RenderUpdateFlag::Path)
         renderDataShape->tesselate(mDevice, mQueue, rshape);
-
-    if (flags & RenderUpdateFlag::Stroke)
         renderDataShape->stroke(mDevice, mQueue, rshape);
+    }
 
-    // setup shape fill properties
-    renderDataShape->mRenderSettingsShape.update(mQueue, rshape.fill, flags, transform, mViewMatrix, rshape.color,
-        mPipelineLinear, mPipelineRadial, mPipelineSolid);
+     // update paint settings
+    if (flags & (RenderUpdateFlag::Transform | RenderUpdateFlag::Blend)) {
+        WgShaderTypeMat4x4f modelMat(transform);
+        WgShaderTypeBlendSettigs blendSettigs(ColorSpace::ABGR8888);
```

Comment:
For render shape colorspace is not nessesary, that`s a setting for picure entity. So it can be as some default value there. But we can pass glocal color space setting

Reply:
This colorspace is not for blending? when rendering the shape, probably it needs blennding option. My point is that the  colorspace is fixed in your code. Probably, you need to revisit this later.

Reply:
Blending settings will user also for opacity settings and maybe for composit.